subprojects/core-api/src/main/java/org/gradle/api/tasks/TaskInputFilePropertyBuilder.java

```diff
@@ -79,14 +79,4 @@
      */
     @Incubating
     TaskInputFilePropertyBuilder ignoreEmptyDirectories();
-
-    /**
-     * Sets whether directories should be considered during up-to-date checks.  Defaults to false.
-     *
-     * See {@link #ignoreEmptyDirectories()}.
-     *
-     * @since 6.8
-     */
-    @Incubating
-    TaskInputFilePropertyBuilder ignoreEmptyDirectories(boolean ignoreEmptyDirectories);
```

Comment:
Why remove this? We have the same for `optional()` and `skipWhenEmpty()`.

Reply:
Yeah, that's the only reason I had put it in there in the first place - because it was there for other things.  I removed it because we changed `DirectorySensitivity` to be either `DEFAULT` or `IGNORE`, so it no longer made sense.  We no longer have something that represents that you explicitly want to fingerprint directories (i.e. `ignoreEmptyDirectories(false)`).

For instance, if someone did something like `ignoreDirectories(false).withNormalizer(CLASSPATH)` what should we do?  Ideally we would throw an error, but how do we tell when the value can only be `DEFAULT` or `IGNORE`?  So we could either silently allow nonsensical behavior, which seems wrong; or we could just track the fact that you tried to set `ignoreDirectories()` at all with a non-configurable normalizer, which means we would error on `ignoreDirectories(true)` as well, which also seems wrong; or we could re-introduce the "I explicitly set it to fingerprint directories" state (which we just removed); or we just remove this method which doesn't make complete sense anyways (i.e. there is no normalizer where it would be _necessary_ to say `ignoreEmptyDirectories(false)`).

Reply:
Another option here, I suppose would be to expose `DirectorySensitvity` on the public api and allow users to set it that way, but that feels kind of gross, too.

Reply:
Hm. I don't think we've been very rigorous with this API in the past, e.g. `skipWhenEmpty(true)` can be applied to a `file()` input where we don't actually take it into account.

To me calling `ignoreDirectories(false)` _could_ mean not to specifically ignore directories (i.e. `DEFAULT`).

As to `ignoreDirectories(false).withNormalizer(CLASSPATH)` our current rule is to use the latest specified values. We allow for example to do `withPathSensitivity(NONE).withNormalizer(CLASSPATH).withPathSensitivity(ABSOLUTE)`.

So I'd just treat this as a mutable thing where calling one method might override the result of a previous call.

Reply:
Ok, maybe I'm being too strict about not allowing thie sort of meaningless configuration.  I've re-introduced the `ignoreEmptyDirectories(boolean)` method.  If someone were to say `ignoreDirectories(false).withNormalizer(CLASSPATH)`, it will just accept this and operate as normal.